Panic in `gh pr view --comments` due to nil pointer dereference in project items

### Describe the bug

When running `gh pr view <number> --comments`, the CLI panics with a `runtime error: invalid memory address or nil pointer dereference`.

This occurs in `api/queries_projects_v2.go` when processing `projectItems`. It appears the GraphQL API is returning null nodes in the `projectItems` connection, which the client does not check before dereferencing.

I observed this even when using a personal access token with full read-only access to all scopes, so I don't think this is a permission issue.

### Affected version

```
$ gh version
gh version 2.83.1-68-g51dfeeaa7 (2025-12-16)
https://github.com/cli/cli/releases/latest
```

### Steps to reproduce the behavior

```shell
$ gh pr view 5475 --repo kubernetes/enhancements
⡿panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x84fbaa]

goroutine 102 [running]:
github.com/cli/cli/v2/api.ProjectsV2ItemsForPullRequest(0xc00006ff60, {0x25b5320, 0xc0002490b0}, 0xc00085d008)
        github.com/cli/cli/v2/api/queries_projects_v2.go:179 +0x3aa
github.com/cli/cli/v2/pkg/cmd/pr/shared.(*finder).Find.func5()
        github.com/cli/cli/v2/pkg/cmd/pr/shared/finder.go:295 +0x32
golang.org/x/sync/errgroup.(*Group).Go.func1()
        golang.org/x/sync@v0.19.0/errgroup/errgroup.go:93 +0x50
created by golang.org/x/sync/errgroup.(*Group).Go in goroutine 1
        golang.org/x/sync@v0.19.0/errgroup/errgroup.go:78 +0x95
```

### Expected vs actual behavior

The `gh` CLI returns PR details.

See above for the actual behavior.

Thanks, @chrishenzie!  It appears the `projectItems` field is returning a `null` node, which causes the panic trying to dereference within the following loop:

```json
{
  "data": {
    "repository": {
      "pullRequest": {
        "projectItems": {
          "totalCount": 1,
          "nodes": [
            null
          ],
          "pageInfo": {
            "hasNextPage": false,
            "endCursor": "MQ"
          }
        }
      }
    }
  }
}
```

https://github.com/cli/cli/blob/baf629918338c8fed6570a2252118f5630362282/api/queries_projects_v2.go#L170-L189

<details>
<summary>
  <b>Expand for the full debug log entry</b>
</summary>
<p>

```
* Request at 2025-12-17 11:09:43.485672464 -0800 PST m=+1.034953786
* Request to https://api.github.com/graphql
> POST /graphql HTTP/1.1
> Host: api.github.com
> Accept: application/vnd.github.merge-info-preview+json, application/vnd.github.nebula-preview
> Authorization: token ████████████████████
> Content-Length: 476
> Content-Type: application/json
> Graphql-Features: merge_queue
> Time-Zone: America/Los_Angeles
> User-Agent: GitHub CLI 2.83.2

GraphQL query:
query PullRequestProjectItems($endCursor:String$name:String!$number:Int!$owner:String!){repository(owner: $owner, name: $name){pullRequest(number: $number){projectItems(first: 100, after: $endCursor){totalCount,nodes{id,project{id,title},status:fieldValueByName(name: "Status"){... on ProjectV2ItemFieldSingleSelectValue{optionId,name}}},pageInfo{hasNextPage,endCursor}}}}}
GraphQL variables: {"endCursor":null,"name":"enhancements","number":5475,"owner":"kubernetes"}

⢿< HTTP/2.0 200 OK
< Access-Control-Allow-Origin: *
< Access-Control-Expose-Headers: ETag, Link, Location, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Used, X-RateLimit-Resource, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval, X-GitHub-Media-Type, X-GitHub-SSO, X-GitHub-Request-Id, Deprecation, Sunset
< Content-Security-Policy: default-src 'none'
< Content-Type: application/json; charset=utf-8
< Date: Wed, 17 Dec 2025 19:09:43 GMT
< Github-Authentication-Token-Expiration: 2026-03-16 11:17:08 -0700
< Referrer-Policy: origin-when-cross-origin, strict-origin-when-cross-origin
< Server: github.com
< Strict-Transport-Security: max-age=31536000; includeSubdomains; preload
< Vary: Accept-Encoding, Accept, X-Requested-With
< X-Content-Type-Options: nosniff
< X-Frame-Options: deny
< X-Github-Media-Type: github.v4; param=merge-info-preview.nebula-preview; format=json
< X-Github-Request-Id: 3809:1B5F77:4E84C8E:5043225:6942FFF7
< X-Ratelimit-Limit: 5000
< X-Ratelimit-Remaining: 4992
< X-Ratelimit-Reset: 1766002037
< X-Ratelimit-Resource: graphql
< X-Ratelimit-Used: 8
< X-Xss-Protection: 0

{
  "data": {
    "repository": {
      "pullRequest": {
        "projectItems": {
          "totalCount": 1,
          "nodes": [
            null
          ],
          "pageInfo": {
            "hasNextPage": false,
            "endCursor": "MQ"
          }
        }
      }
    }
  }
}

* Request took 395.025616ms
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x842fca]

goroutine 116 [running]:
github.com/cli/cli/v2/api.ProjectsV2ItemsForPullRequest(0xc000e8df60, {0x25a40e8, 0xc000b50ed0}, 0xc0006b3808)
        /home/runner/work/cli/cli/api/queries_projects_v2.go:179 +0x3aa
github.com/cli/cli/v2/pkg/cmd/pr/shared.(*finder).Find.func5()
        /home/runner/work/cli/cli/pkg/cmd/pr/shared/finder.go:295 +0x32
golang.org/x/sync/errgroup.(*Group).Go.func1()
        /home/runner/go/pkg/mod/golang.org/x/sync@v0.18.0/errgroup/errgroup.go:93 +0x50
created by golang.org/x/sync/errgroup.(*Group).Go in goroutine 1
        /home/runner/go/pkg/mod/golang.org/x/sync@v0.18.0/errgroup/errgroup.go:78 +0x95
```

</p>
</details>

I've opened https://github.com/cli/cli/pull/12356 as I suspect permissions are preventing this v2 project items to being returned, requiring the `gh` code to skip them gracefully like we do in a few other places.

--- a/api/queries_projects_v2.go
+++ b/api/queries_projects_v2.go
@@ -106,6 +106,9 @@ func ProjectsV2ItemsForIssue(client *Client, repo ghrepo.Interface, issue *Issue
 			return err
 		}
 		for _, projectItemNode := range query.Repository.Issue.ProjectItems.Nodes {
+			if projectItemNode == nil {
+				continue
+			}
 			items.Nodes = append(items.Nodes, &ProjectV2Item{
 				ID: projectItemNode.ID,
 				Project: ProjectV2ItemProject{
@@ -175,6 +178,9 @@ func ProjectsV2ItemsForPullRequest(client *Client, repo ghrepo.Interface, pr *Pu
 		}
 
 		for _, projectItemNode := range query.Repository.PullRequest.ProjectItems.Nodes {
+			if projectItemNode == nil {
+				continue
+			}
 			items.Nodes = append(items.Nodes, &ProjectV2Item{
 				ID: projectItemNode.ID,
 				Project: ProjectV2ItemProject{

--- a/api/queries_projects_v2_test.go
+++ b/api/queries_projects_v2_test.go
@@ -129,6 +129,17 @@ func TestProjectsV2ItemsForIssue(t *testing.T) {
 			},
 			expectError: true,
 		},
+		{
+			name: "skips null project items for issue",
+			httpStubs: func(reg *httpmock.Registry) {
+				reg.Register(
+					httpmock.GraphQL(`query IssueProjectItems\b`),
+					httpmock.GraphQLQuery(`{"data":{"repository":{"issue":{"projectItems":{"totalCount":1,"nodes":[null]}}}}}`,
+						func(query string, inputs map[string]interface{}) {}),
+				)
+			},
+			expectItems: ProjectItems{},
+		},
 	}
 
 	for _, tt := range tests {
@@ -185,6 +196,17 @@ func TestProjectsV2ItemsForPullRequest(t *testing.T) {
 				)
 			},
 			expectError: true,
+		},
+		{
+			name: "skips null project items for pull request",
+			httpStubs: func(reg *httpmock.Registry) {
+				reg.Register(
+					httpmock.GraphQL(`query PullRequestProjectItems\b`),
+					httpmock.GraphQLQuery(`{"data":{"repository":{"pullRequest":{"projectItems":{"totalCount":1,"nodes":[null]}}}}}`,
+						func(query string, inputs map[string]interface{}) {}),
+				)
+			},
+			expectItems: ProjectItems{},
 		},
 		{
 			name: "retrieves project items that have status columns",

